apps/dolly-frontend/src/main/js/src/components/fagsystem/tpsf/form/personinformasjon/partials/sikkerhetstiltak/Sikkerhetstiltak.tsx

```diff
@@ -27,28 +32,57 @@ type Option = {
 export const Sikkerhetstiltak = ({ formikBag }: SikkerhetstiltakProps) => {
 	const opts = useContext(BestillingsveilederContext)
 
+	const navEnheter = SelectOptionsOppslag.hentNavEnheter()
+	const navEnheterOptions = SelectOptionsOppslag.formatOptions('navEnheter', navEnheter)
+
 	const paths = {
-		typeSikkerhetTiltak: 'tpsf.typeSikkerhetTiltak',
-		beskrSikkerhetTiltak: 'tpsf.beskrSikkerhetTiltak',
-		sikkerhetTiltakDatoFom: 'tpsf.sikkerhetTiltakDatoFom',
-		sikkerhetTiltakDatoTom: 'tpsf.sikkerhetTiltakDatoTom',
+		rootPath: 'pdldata.person.sikkerhetstiltak[0]',
+		tiltakstype: 'pdldata.person.sikkerhetstiltak[0].tiltakstype',
+		kontaktperson: 'pdldata.person.sikkerhetstiltak[0].kontaktperson',
+		gyldigFraOgMed: 'pdldata.person.sikkerhetstiltak[0].gyldigFraOgMed',
+		gyldigTilOgMed: 'pdldata.person.sikkerhetstiltak[0].gyldigTilOgMed',
+		tpsMessagingRootPath: 'tpsMessaging.sikkerhetstiltak',
 	}
 
 	const indexBeskrSikkerhetTiltak = 7
 
 	const handleSikkerhetstiltakChange = (option: Option) => {
-		formikBag.setFieldValue(paths.typeSikkerhetTiltak, option.value)
-		formikBag.setFieldValue(
-			paths.beskrSikkerhetTiltak,
-			option.label === 'Opphørt' ? option.label : option.label.substring(indexBeskrSikkerhetTiltak)
+		handleValueChange(option.value, 'tiltakstype')
+		handleValueChange(
+			option.label === 'Opphørt' ? option.label : option.label.substring(indexBeskrSikkerhetTiltak),
+			'beskrivelse'
 		)
 	}
 
+	const handleValueChange = (value: string, path: string) => {
+		formikBag.setFieldValue(`${paths.rootPath}.${path}`, value)
+		formikBag.setFieldValue(`${paths.tpsMessagingRootPath}.${path}`, value)
+	}
+
+	const gyldigFomErDagensDato = () => {
+		return isToday(_get(formikBag.values, 'pdldata.person.sikkerhetstiltak[0].gyldigFraOgMed'))
+	}
+
+	const [randomNavUsers, setRandomNavUsers] = useState([])
+	const [personident, setPersonident] = useState(
+		_get(formikBag.values, 'pdldata.person.sikkerhetstiltak[0].kontaktperson.personident')
+	)
+
+	useEffect(() => {
+		setRandomNavUsers(genererTilfeldigeNavPersonidenter())
+	}, [])
+
 	return (
 		<Vis attributt={Object.values(paths)} formik>
 			<div className="flexbox--flex-wrap">
```

Comment:
Denne skulle kanskje vært wrappet i en FormikDollyFieldArray? Slik som resten av pdl-attributtene, sånn at man kan legge til flere. Eller er det slik at sikkerhetstiltak er et unntak? 😁

Reply:
Hmm, tps-messaging støtter bare innsending av et objekt og ikke liste på denne så tok utgangspunkt i det 😅  Isåfall må det støttes liste på pdl og bare sendes en til tps-messaging, det blir kanskje litt knot? 🤪

Reply:
Haha, jepp, det blir knot! 😅  Men om pdl støtter innsending av flere burde vi kanskje gjøre det? Mulig @krharum har en mening om dette?

Reply:
Ja stemmer at TPS bare støtter et sikkerhetstiltak. Så lenge TPS er med stemmer jeg for at vi begrenser det til ET selv om det ser ut til at PDL støtter flere.